src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+                }
+            }
+
+            data_str = dict_to_json_str(data)
+            await self.wallet_state_manager.create_action(
+                name="request_generator",
+                wallet_id=self.wallet_info.id,
+                type=self.wallet_info.type,
+                callback="generator_received",
+                done=False,
+                data=data_str,
+            )
+
+    # This should basically never be called as we don't want to receive ID coins from somebody else
+    async def search_for_parent_info(
+        self, block_program: Program, removals: List[Coin]
+    ) -> bool:
+
+        """
+        Returns an error if it's unable to evaluate, otherwise
+        returns a list of NPC (coin_name, solved_puzzle_hash, conditions_dict)
+        """
+        cost_sum = 0
+        try:
+            cost_run, sexp = block_program.run_with_cost([])
+            cost_sum += cost_run
+        except EvalError:
+            return False
+
+        for name_solution in sexp.as_iter():
+            _ = name_solution.as_python()
+            if len(_) != 2:
+                return False
+            if not isinstance(_[0], bytes) or len(_[0]) != 32:
+                return False
+            coin_name = bytes32(_[0])
+            if not isinstance(_[1], list) or len(_[1]) != 2:
+                return False
+            puzzle_solution_program = name_solution.rest().first()
+            puzzle_program = puzzle_solution_program.first()
+            try:
+                error, conditions_dict, cost_run = conditions_dict_for_solution(
+                    puzzle_solution_program
+                )
+                cost_sum += cost_run
+                if error:
+                    return False
+            except clvm.EvalError:
+
+                return False
+            if conditions_dict is None:
+                conditions_dict = {}
+
+            if ConditionOpcode.CREATE_COIN in conditions_dict:
+                created_output_conditions = conditions_dict[ConditionOpcode.CREATE_COIN]
+            else:
+                continue
+            for cvp in created_output_conditions:
+                result = await self.wallet_state_manager.puzzle_store.wallet_info_for_puzzle_hash(
+                    cvp.var1
+                )
+                if result is None:
+                    continue
+
+                wallet_id, wallet_type = result
+                if wallet_id != self.wallet_info.id:
+                    continue
+
+                coin = None
+                for removed in removals:
+                    if removed.name() == coin_name:
+                        coin = removed
+                        break
+
+                if coin is not None:
+                    if did_wallet_puzzles.check_is_did_puzzle(puzzle_program):
+                        inner_puzzle_hash = (
+                            did_wallet_puzzles.get_innerpuzzle_from_puzzle(
+                                puzzle_program
```

Comment:
Using the `get_` prefix is common practice in Java and probably some other languages, but it's not typically used in python and seems overly verbose to me.